protocol/message.go

```diff
@@ -211,6 +220,45 @@ func isRTL(s string) bool {
 		first == '\u200f'
 }
 
+// parseImage check the message contains an image, and if so
+// it creates the a base64 encoded version of it.
+func (m *Message) parseImage() error {
+	if m.ContentType != protobuf.ChatMessage_IMAGE {
+		return nil
+	}
+	image := m.GetImage()
+	if image == nil {
+		return errors.New("image empty")
+	}
+
+	payload := image.Payload
+
+	e64 := base64.StdEncoding
+
+	maxEncLen := e64.EncodedLen(len(payload))
+	encBuf := make([]byte, maxEncLen)
+
+	e64.Encode(encBuf, payload)
+	var mime string
+
+	switch image.Type {
+	case protobuf.ImageMessage_PNG:
+		mime = "png"
+	case protobuf.ImageMessage_JPEG:
+		mime = "jpeg"
+	case protobuf.ImageMessage_WEBP:
+		mime = "webp"
+	case protobuf.ImageMessage_GIF:
+		mime = "gif"
+	default:
+		return errors.New("image format not supported")
+	}
```

Comment:
Is it possible to have this code be a function? That would allow for this functionality to be testable.

Something like

```go
func getImageMessageMIME(i protobuf.ImageMessage) (string, error) {
	switch i.Type {
	case protobuf.ImageMessage_PNG:
		return "png", nil
	case protobuf.ImageMessage_JPEG:
		return "jpeg", nil
	case protobuf.ImageMessage_WEBP:
		return "webp", nil
	case protobuf.ImageMessage_GIF:
		return "gif", nil
	default:
		"", return errors.New("image format not supported")
	}
}
```

Usage:

```go
mime, err := getImageMessageMIME(image)
```